src/shared/nested-dropdown-list/types.ts

```diff
@@ -0,0 +1,84 @@
+import { Option } from "../../input-nested-select";
+
+export type TruncateType = "middle" | "end";
+export type ItemsLoadStateType = "loading" | "fail" | "success";
+export type Mode = "default" | "expand" | "collapse";
+
+export interface ListItemRenderArgs {
+    selected: boolean;
+}
+
+export interface DropdownDisplayProps<V1, V2, V3> {
+    /** Control the dropdown list when it is first opened */
+    mode?: Mode;
+    /** Expand the recursively selected key in dropdown list when it is first opened */
+    selectedKeyPath: string[];
```

Comment:
ideally this nested dropdown should be able to handle multi-select too, so this would be a `string[][]`

but leave it for now since this can be reworked when working on the nested multi-select